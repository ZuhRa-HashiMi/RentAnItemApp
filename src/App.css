@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@300;500;600&family=Roboto&display=swap');

*,
::after,
::before {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: 'Poppins', sans-serif;
  font-weight: 100;
}

body {
  background-color: #f6f3f8;
}

:root {
  --navbar-icon-color: #a7a7a7;
  --navbar-link-color: #7e7d7f;
  --orange: #ed6b3b;
}

a {
  color: inherit;
  text-decoration: none;
}

.row {
  display: flex;
}

.column {
  display: flex;
  flex-direction: column;
}

.flex-center {
  display: flex;
  justify-content: center;
  align-items: center;
}

.space-between {
  display: flex;
  justify-content: space-between;
}

.max-content-flex {
  flex: 1;
}

.wrap {
  flex-wrap: wrap;
}

.section {
  padding-top: 80px;
}

.profile-photo {
  aspect-ratio: 1 / 1;
  width: 2rem;
  border-radius: 50%;
}

.mob-profile-photo {
  width: 4rem;
}

.mob-username {
  font-size: 1.3rem;
  margin: 1rem 0 0 0;
}

.active {
  font-family: 'Roboto', sans-serif;
}

.divider {
  opacity: 0.3;
  width: 70%;
  margin: 3rem 0 1rem 0;
}

.carousel.carousel-slider .control-arrow {
  top: 45%;
  background-color: black;
  border-radius: 50%;
  bottom: auto;
  padding: 0.5rem;
}

li.selected img {
  box-shadow: 2px 10px 22px -7px rgba(0, 0, 0, 0.75);
  -webkit-box-shadow: 2px 10px 22px -7px rgba(0, 0, 0, 0.75);
  -moz-box-shadow: 2px 10px 22px -7px rgba(0, 0, 0, 0.75);
  border: 1px solid rgba(0, 0, 0, 0.3);
}

li.slide :not(.selected) img {
  transform: scale(0.8);
  transform: translate(2px, -5%);
  padding: 4rem;
}

li.slide div {
  background-color: transparent;
}

.carousel-root {
  margin: 2rem;
  background-color: #fff;
}

.carousel .control-dots {
  translate: 0 1rem;
}

.caroselImage {
  width: 100%;
}

.btn-default {
  background-color: var(--orange);
  color: white;
  padding: 0.5rem 1rem;
  border-radius: 0.2rem;
  font-size: 1rem;
  border: 0;
}

.btn-default:focus {
  background-color: #d5582a;
}

.empty-list {
  width: 100%;
  min-height: 350px;
  text-align: center;
}

.add-favorite {
  padding: 1rem;
}

.add-house-section {
  background: url(./assets/images.jpeg) no-repeat;
  background-size: cover;
}

.backdrop {
  background-color: #ffffffb5;
  backdrop-filter: blur(1px);
  min-height: 100vh;
  text-align: center;
}

.backdrop p {
  width: 50%;
  font-weight: 300;
  font-size: 0.8rem;
}

form {
  margin: 2rem;
  gap: 1rem;
  width: 100%;
  align-items: center;
}

.input-text,
.input-textarea {
  border: 1px solid rgba(0, 0, 0, 0.359);
  border-radius: 1.5rem;
  padding: 0.5rem 1rem;
  width: 50%;
  background-color: transparent;
  outline-color: var(--orange);
  font-weight: 500;
}

.input-textarea {
  border-radius: 0.5rem;
  height: 7rem;
}

.btn {
  border: none;
  border-radius: 1.5rem;
  padding: 0.5rem 1.3rem;
  background-color: #ec6232;
  outline: none;
  font-weight: 500;
  color: #f6f3f8;
  box-shadow: -1px -1px 53px 0 #ed6b3b;
  -webkit-box-shadow: -1px -1px 53px 0 #ed6b3b;
  -moz-box-shadow: -1px -1px 53px 0 #ed6b3b;
}

.btn:active {
  background-color: #e95321;
}

.btn:is(:hover, :active) {
  transform: scale(1.1);
  background-color: #ec6232;
  box-shadow: -1px -1px 53px 0 #ed6b3b;
  -webkit-box-shadow: -1px -1px 53px 0 #ed6b3b;
  -moz-box-shadow: -1px -1px 53px 0 #ed6b3b;
}

.flash-message {
  color: #d5582a;
  font-size: 1rem;
  font-weight: 500;
}

@media screen and (max-width: 764px) {
  .desktop-only {
    display: none;
  }
}
