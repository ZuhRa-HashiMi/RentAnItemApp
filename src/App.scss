
.header {
  width: 100%;
  height: 300%;
  background: url("./assets/home.jpg") bottom center no-repeat;
  background-size: cover;
  padding: 100px;

  &-content {
    width: 215px;
    color: #fff;

    &__title {
      font-size: 22px;
      font-weight: 700;
      text-shadow: 1px 2px 2px rgba($color: #000, $alpha: 0.6);
    }

    &__desc {
      font-weight: 100;
      text-shadow: 0 1px 1px rgba($color: #000, $alpha: 0.4);
    }
  }
}

.btn-oval {
  border-radius: 35px !important;
  background-color: #ea531c !important;
  border-color: #ea531c !important;
  color: #fff !important;
  padding: 5px 20px;
}

.btn-menu {
  border-radius: 35px !important;
  background-color: #787877 !important;
  border-color: #787877 !important;
  color: #fff !important;
  padding: 5px 30px;
}

.auth_links {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: 0;
  padding: 0;
  font-size: 20px;
}

.white {
  color: #fff !important;
}


.container {
  max-width: 768px !important;
  padding: 0 !important;
}

.slack {
  margin: 0 auto;
}

.logo {
  width: 150px;
}

.about-section {
  padding: 30px;
  background-color: #fff;

  &__sub {
    text-transform: uppercase;
    color: #b9b6b6;
    text-align: center;
    font-size: 13px;
  }

  &__primary,
  &__left-title {
    text-align: center;
    font-size: 25px;
    font-weight: 300;
  }

  &__left-title {
    text-align: left;
  }

  &__desc {
    font-weight: 100;
    font-size: 14px;
    color: #000;
  }
}

.feature-section {
  background-color: #e8f0ff;
}

.call-action {
  height: 150px;
  background: url("./assets/profile-photo.jpeg") top center no-repeat;
  color: #e8f0ff;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
.custom-navbar{
  background-color: transparent;

}

.footer {
  color: #b9b6b6;
  text-align: center;
  font-size: small;
}

.login-box {
  display: flex;
  height: 100vh;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  text-align: center;
  background-color: rgba(255, 255, 255, 0.8);
  position: relative;
}

.login-box::after {
  content: "";
  background: url("./assets/profile-photo.jpeg") bottom center no-repeat;
  background-size: cover;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  position: absolute;
  z-index: -1;
}

.form-control {
  border-radius: 20px !important;
  padding: 5px 10px;
}

.modal-popup {
  overflow: hidden;
  padding: 5px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
/* Media queries for responsiveness */
@media screen and (max-width: 767px) {
  .header {
    height: 300%;
    padding: 40px;

    &-content {
      width: 100%;
      max-width: 80%;

      &__title {
        font-size: 16px;
      }

      &__desc {
        font-size: 12px;
      }
    }
  }

  .btn-oval {
    padding:0 25px;
  }


  .auth_links {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    font-size: 20px;
  
    .nav-link-space {
      margin-right: 10px; /* Adjust the margin value as needed */
    }
  }
  .logo {
    width: 100px;
  }

  .about-section {
    padding: 10px;

    &__sub {
      font-size: 10px;
    }

    &__primary,
    &__left-title {
      font-size: 18px;
    }

    &__desc {
      font-size: 12px;
    }
  }

  /* Additional styles for small screens */
}

/* Media queries for larger screens if needed */
@media screen and (max-width: 575px) {
  /* Styles for iPhone */
  .header {
  height: 150%;
  padding: 50px;
  &-content {
  width: 150px;
  &__title {
  font-size: 18px;
  }
  &__desc {
  font-size: 12px;
  }
  }
  }
  .btn-oval {
  padding: 5px 10px;
  }
  .btn-menu {
  padding: 5px 5px;
  }
  .auth_links {
  font-size: 16px;
  }
  .logo {
  width: 80px;
  }
  .about-section {
  &__primary,
  &__left-title {
  font-size: 22px;
  }
  &__desc {
  font-size: 12px;
  }
  }
  .call-action {
  height: 100px;
  }
  .footer {
  font-size: x-small;
  }
  }
  
  @media screen and (min-width: 576px) and (max-width: 767px) {
  /* Styles for Samsung */
  .header {
  height: 200%;
  padding: 75px;
  &-content {
  width: 180px;
  &__title {
  font-size: 20px;
  }
  &__desc {
  font-size: 14px;
  }
  }
  }
  .btn-oval {
  padding: 5px 15px;
  }
  .btn-menu {
  padding: 5px 7.5px;
  }
  .auth_links {
  font-size: 18px;
  }
  .logo {
  width: 90px;
  }
  .about-section {
  &__primary,
  &__left-title {
  font-size: 23px;
  }
  &__desc {
  font-size: 13px;
  }
  }
  .call-action {
  height: 125px;
  }
  .footer {
  font-size: small;
  }
  }