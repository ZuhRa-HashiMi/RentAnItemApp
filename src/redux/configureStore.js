/* eslint-disable no-underscore-dangle */

import { configureStore } from '@reduxjs/toolkit';

const store = configureStore({
  reducer: { },
});

export default store;
