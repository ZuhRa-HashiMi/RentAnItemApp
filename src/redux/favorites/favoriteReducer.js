import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';

export const fetchFavorites = createAsyncThunk('rent-house/favorites', async (userId) => {
  const result = await fetch(`http://localhost:3000/api/v1/users/${userId}/favorite_houses`)
  return result.json()
})

const favoriteReducer = createSlice({
  name: "favorites",
  initialState: {
    favorites: [],
  },
  extraReducers: (builder) => {
    builder.addCase(fetchFavorites.fulfilled, 
      (state, action) => {
        return ({...state, favorites: [...action.payload]})}
    )
  }
});

export default favoriteReducer.reducer;
