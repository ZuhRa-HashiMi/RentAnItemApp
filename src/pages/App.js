import React from 'react';
import '../App.scss';
import { Container } from 'react-bootstrap';
import Header from '../componenets/Header';
import FeatureSection from '../componenets/FeatureSection';
import Footer from '../componenets/Footer';
import '../landing.css';

function App() {
  return (
    <Container>
      <Header />
      <FeatureSection />
      <Footer />
    </Container>
  );
}

export default App;
