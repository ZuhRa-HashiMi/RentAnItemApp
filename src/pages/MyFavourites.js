import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import { deleteFavorite, fetchFavorites } from '../redux/favorites/favoriteReducer';
import House from '../componenets/house/House';
import Navbar from '../componenets/navbar/Navbar';

const MyFavourites = () => {
  const currentUser = useSelector((state) => state.favorite.user);
  const { favorites, status } = useSelector((state) => state.favorite);
  const dispatch = useDispatch();
  useEffect(() => {
    console.log(currentUser)
    if (status === 'idle') {
      dispatch(fetchFavorites(currentUser.id));
    }
  });

  const handleRemove = (houseID) => {
    dispatch(deleteFavorite({ userID: currentUser.id, houseID }));
  };

  return (
    <>
      <Navbar title="My Favorites" />
      <section className="section">
        {favorites?.length === 0 && (<p className="flex-center empty-list">No Houses: List Empty</p>)}
        <ul className="flex-center wrap">
          {favorites.map((house) => (
            <li key={house.id} className="flex-center house-wrapper">
              <House data={house} handleRemove={handleRemove} />
            </li>
          ))}
        </ul>
      </section>
    </>
  );
};

export default MyFavourites;
