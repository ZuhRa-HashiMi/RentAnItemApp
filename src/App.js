import { Route, Routes } from 'react-router-dom';
import './App.css';
// import Navbar from './componenets/Navbar';
import Home from './pages/Homepage';
import MyFavourites from './pages/MyFavourites';
import AddHouse from './pages/AddHouse';
import DeleteHouse from './pages/DeleteHouse';
import HouseDetails from './pages/HouseDetails';
import Header from './componenets/Header';
import FeatureSection from './componenets/FeatureSection';
import Footer from './componenets/Footer';

function App() {
  return (
    <div className="App">
      {/* <Navbar /> */}
      <Header />
      <FeatureSection />
      <Footer />
      <Routes>

        <Route path="/" element={<Home />} />
        <Route path="/MyFavourites" element={<MyFavourites />} />
        <Route path="/AddHouse" element={<AddHouse />} />
        <Route path="/DeleteHouse" element={<DeleteHouse />} />
        <Route path="/HouseDetails" element={<HouseDetails />} />
      </Routes>
    </div>
  );
}

export default App;
