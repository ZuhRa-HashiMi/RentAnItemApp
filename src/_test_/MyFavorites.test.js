import renderer, { act } from 'react-test-renderer';
import '@testing-library/jest-dom';
import React from 'react';
import { Provider } from 'react-redux';
import { fetchFavoriteHouseData, fetchHouseData, testListFavoriteHouses } from './setupTests';
import configureStore from '../redux/configureStore';
import MyFavourites from '../pages/MyFavourites';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import App from '../App';
import { fetchFavorites } from '../redux/favorites/favoriteReducer';

const store = configureStore;
fetchHouseData()
fetchFavoriteHouseData(1);

describe('test render', () => {
  test('List Favorites should match snapshoot', () => {
    const tree = renderer.create(
      <React.StrictMode>
        <Provider store={store}>
          <MyFavourites />
        </Provider>
      </React.StrictMode>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
