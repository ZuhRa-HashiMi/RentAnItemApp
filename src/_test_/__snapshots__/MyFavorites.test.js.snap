// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render List Favorites should match snapshoot 1`] = `
Array [
  <h1
    className="flex-center"
  >
    Favourites
  </h1>,
  <ul
    className="flex-center wrap"
  >
    <p
      className="flex-center empty-list"
    >
      No favorites: List Empty
    </p>
  </ul>,
]
`;
